pkg/healthz/healthz.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2023 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package healthz
+
+import (
+	"fmt"
+	"net/http"
+	"sort"
+	"sync"
+
+	xhttp "github.com/intel/nri-resmgr/pkg/http"
+	logger "github.com/intel/nri-resmgr/pkg/log"
+)
+
+var (
+	lock     sync.Mutex
+	checkers = map[string]CheckFn{}
+	sorted   []string
+	// our logger instance
+	log = logger.NewLogger("health-check")
+)
+
+type CheckFn func() (status Status, details error)
+
+// Status describes the health of a component or the whole.
+type Status int
+
+const (
+	// just an example, we need to figure out what/if granularity makes any sense
+	Healthy Status = iota
+	Degraded
+	NonFunctional
+	NuclearMeltdown
```

Comment:
I think we could update these to a more final granularity already in the initial commit. At least I'd take out the `NuclearMeltdown`.

Reply:
ah good point. Let me clean that up.